### PR TITLE
Fix Pusher Re-Authentication bug

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
Well explained in the linked issue.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/152022

### Tests
**Testing subscribing with bad session**
1. Invalidate your authToken by messing with it in `localStorage`
1. Filter in the JS console logs for `Pusher`
1. Tap on a chat in the LHN
1. Verify that your logs look like this and that Pusher has successfully resubscribed to all channels
Screen Shot 2021-01-22 at 1 47 54 PM

**Testing loss of session and connectivity** 
1. Invalidate your authToken by messing with it in `localStorage`
2. Turn off your WiFi
3. Filter in the JS console logs for `Pusher`
4. Wait until you see this log line in the JS console
Screen Shot 2021-01-22 at 1 52 17 PM
5. Turn WiFi back on
6. Don't touch anything! Wait and then verify that you are eventually resubscribed to Pusher
Screen Shot 2021-01-22 at 1 52 49 PM

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/libs/API.js [hunk 1]
```diff
@@ -122,30 +122,8 @@ function handleExpiredAuthToken(originalCommand, originalParameters, originalTyp
     isAuthenticating = true;
 
     // eslint-disable-next-line no-use-before-define
```
Comment (Gonals) [inline src/libs/API.js:124]: Do we still need this?

**Review by Gonals — COMMENTED** [review]
LGTM. One minor question!

File: src/libs/API.js (hunk 1 above)
Comment (marcaaron) [reply src/libs/API.js:124]: Yeah I think we do since `reauthenticate()` is still defined after this function.

**Comment by marcaaron** [comment]
@Gonals Thanks for the review!

**Review by Gonals — APPROVED** [review]

**Pull request merged**